doc/proxy-clients.rst

```diff
@@ -101,12 +101,24 @@ You can also pass some options to the Varnish client:
   which tags to invalidate when sending invalidation requests to the caching
   proxy. Make sure that your :ref:`Varnish configuration <varnish_tagging>`
   corresponds to the header used here;
+* ``tag_mode`` (ban): Makes it possible to invalidate tags
+  using purge when for instance using ``xkey`` instead of ban. Supports three
+  modes: ``ban``, ``purge``, and ``purge_single`` for purge per tag.
```

Comment:
we should list what the modes do. maybe do like this: 
```
You can use the [link:xkey] module for improved cache tag handling. Without xkey, the only
supported mode is ``ban``. With xkey the supported modes are ``purge`` and ``purge_single``.
``purge_single`` sends a separate purge request per tag, which is the only mode supported by
the varnish modules prior to v0.10.2. If you have a newer version of the modules, it is more
efficient to use ``purge``.
```

Reply:
thinking about this some more, i think we should have a separate flag `tag_purge_single: false` that can be set to true and is about legacy modules support. then we can use ban/purge/softpurge|expire as options here. what is the better name, softpurge or expire?

Reply:
and we should say "See below for how to configure your application to support xkey".

Reply:
Ok, but note that #385 was not about letting people configure soft_purge over purge here, it was about the code using FOSHttpCache being able to tell the system if soft purge is ok or not, so I don't see a need for a soft_purge option here.

E.g. On deletion of an entity, soft purge is probably not desirable as it can lead to broken links during stale periods, while a creation or update of a entity can safely just expire whatever needs to be refreshed.

Reply:
but would it not make sense to have vcl handle both cases, and have the application decide whether soft or hard purge should be done?

Reply:
Yes, but I don't think that should be config, rather something application specifies on purge _(purge vs expiry, where the latter allows stalnes by grace or by stale headers depending on proxy)_-